openlibrary/macros/SocialShare.html

```diff
@@ -5,27 +5,76 @@
-  <p class="cta-section-title">$_('Share this book')</p>
-  <ul class="shareLinks-list">
-    $for share_link in share_links:
-      $ track_tag = share_link['text'].replace(' ', '-')
-      <li>
-        <a href="$share_link['url']" class="sansserif large" target="_blank" data-ol-link-track="Share|$track_tag">
-          <img title="Share on $(share_link['text'])" alt="$(share_link['text'])" class="share-link" src="$static_url('images/%s.svg' % share_link['text'].lower())"/>
-        </a>
-        <div class="share-source">$share_link['text']</div>
-      </li>
-
-    <li>
-      $# rm special characters from url such as single quote which break js prompt/iframe
-      $ clean_url = page_url.replace("'", "\\'")
-      $ iframe_html = str(embed_iframe(clean_url)).strip()
-      <a class="embed-work-btn" title="$_('Embed this book in your website')" data-ol-link-track="Share|Embed"
-         onclick="prompt('Copy embed code to clipboard:', '$iframe_html');">
-        <img alt="Embed icon" class="share-link" src="$static_url('images/embed.png')">
+
+
+  <div>
+    <p style="display: inline-block" class="cta-section-title">$_('Share this book')
+
+    </p>
+
+    <ul class="shareLinks-list" style="display: inline-block; margin-left: 50px; margin-bottom: 4px;">
+      <a class="view-modal">
+          <img class="share-btn" src="../../static/images/icons/share.svg"/>
       </a>
-      <div class="share-source">$_('Embed')</div>
-    </li>
-  </ul>
+    </ul>
+
+  </div>
+
 </div>
+
+
+    <div class="popup">
+
+      <div class="content">
+        <header class="header-share">
+          <p>Share via</p>
+            <div class="close">
+              <i class="uil uil-times"></i>
+            </div>
+        </header>
+        <ul class="icons">
+
+          $for share_link in share_links:
+          $ track_tag = share_link['text'].replace(' ', '-')
+
+              <a href="$share_link['url']" class="sansserif large" target="_blank" data-ol-link-track="Share|$track_tag">
+                <img title="Share on $(share_link['text'])" alt="$(share_link['text'])" class="share-link" src="$static_url('images/%s.svg' % share_link['text'].lower())"/>
+              </a>
+
+
+
+        $# rm special characters from url such as single quote which break js prompt/iframe
+        $ clean_url = page_url.replace("'", "\\'")
+        $ iframe_html = str(embed_iframe(clean_url)).strip()
+        <a class="embed-work-btn" title="$_('Embed this book in your website')" data-ol-link-track="Share|Embed"
+          onclick="prompt('Copy embed code to clipboard:', '$iframe_html');">
+          <img alt="Embed icon" class="share-link" src="$static_url('images/embed.png')">
+        </a>
+
+
+        </ul>
+        <p>Or copy link</p>
+        <div class="copy-field">
+          <i class="url-icon uil uil-link"></i>
+          <input type="text" readonly value="$page_url">
+          <button class="copy-button">Copy</button>
+        </div>
+      </div>
+    </div>
```

Comment:
No need to create a new modal component.  Open Library uses [Colorbox](https://www.jacklmoore.com/colorbox/) for our other modals, and it should also be used here.  Colorbox takes an element's ID, and displays that element in a modal.

You can look at our book notes modal for an example, [here](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/macros/NotesModal.html).  You'll see that the notes modal component consists of a container with a link and a `hidden` div containing the modal's content.

In `index.js`, we load the JS for the notes modal if we find any elements with class `notes-modal-link` on the page, [here](https://github.com/internetarchive/openlibrary/blob/e1f8a358390d318c73f27a6dd8a5d208811a79cc/openlibrary/plugins/openlibrary/js/index.js#L257-L273). If there are `notes-modal-link` elements on the page, `initNotesModal` is called.  This calls `addClickListeners`, which displays the modal on `notes-modal-link` clicks.  All of these functions can be found [here](https://github.com/internetarchive/openlibrary/blob/e1f8a358390d318c73f27a6dd8a5d208811a79cc/openlibrary/plugins/openlibrary/js/modals/index.js) for reference.

You can probably place the original share links content inside of a Colorbox modal and solve this issue without much new CSS.